src/modules/universal-discovery/components/finder/finder.component.js

```diff
@@ -124,40 +262,48 @@ export default class FinderComponent extends Component {
      * @memberof FinderComponent
      */
     findLocationChildren({parent, location}) {
+        if (this.props.allowedLocations.length === 1) {
+            return;
+        }
+
+        if (this.state.locationsMap[parent]) {
+            this.updateSelectedBranches(location);
+            this.props.onItemSelect(location);
+
+            return;
+        }
+
         const promise = new Promise(resolve => this.props.findLocationsByParentLocationId(
```

Comment:
Strange line break.

Reply:
It's your pattern. What would be the best solution for you?